Source/Controllers/PHPostListViewController.swift

```diff
@@ -10,55 +10,55 @@ import Cocoa
 import DateTools
 import SwiftyTimer
 
-class PHPostListViewController: NSViewController, NSTableViewDataSource, NSTableViewDelegate, PHDataSourceDelegate, PHLoadingViewDelegate {
+class PHPostListViewController: NSViewController, NSTableViewDataSource, NSTableViewDelegate, PHDataSourceDelegate {
 
     @IBOutlet weak var productTableView: NSTableView!
     @IBOutlet weak var loadingView: PHLoadingView!
     @IBOutlet weak var lastUpdatedLabel: NSTextField!
     @IBOutlet weak var homeButton: PHButton!
     @IBOutlet weak var settingsButton: PHButton!
 
+    private var source = PHPostsDataSource(store: store)
     private var updateUITimer: NSTimer?
-    private var source = PHPostsDataSource(context: PHAppContext.sharedInstance)
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         view.wantsLayer = true
+
         view.layer?.backgroundColor = NSColor.ph_whiteColor().CGColor
 
         productTableView.intercellSpacing = NSSize.zero
 
         source.delegate = self
 
-        loadingView.delegate = self
         loadingView.showState(.Loading)
 
         homeButton.setImages("Icon-product-hunt", highlitedImage: "icon-kitty")
-
-        updateUI()
     }
 
     override func viewWillAppear() {
         super.viewWillAppear()
 
-        updateUITimer = NSTimer.every(15.seconds) { [weak self] in self?.updateUI() }
-
         self.source.loadNewer()
 
         PHScheduleAsSeenAction.performCancel()
+
+        updateUITimer = NSTimer.every(15.seconds) { [weak self] in self?.updateUI() }
```

Comment:
Yes, this updates `timeAgo` label only. Changes in the store does not occur so often and label stops updating.